x-pack/platform/plugins/shared/onechat/public/application/components/tools/tool.tsx

```diff
@@ -0,0 +1,267 @@
+      }
+      if (tool.type === ToolType.builtin) {
+        reset(transformBuiltInToolToFormData(tool));
+      }
+    }
+  }, [tool, reset]);
+
+  const toolFormId = useGeneratedHtmlId({
+    prefix: 'toolForm',
+  });
+
+  const isViewMode = mode === ToolFormMode.View;
+
+  const saveButton = (
+    <EuiButton
+      size="s"
+      type="submit"
+      fill
+      iconType="save"
+      form={toolFormId}
+      disabled={Object.keys(errors).length > 0 || isSubmitting}
+      isLoading={isSubmitting}
+      css={css`
+        width: 124px;
+      `}
+    >
+      {labels.tools.saveButtonLabel}
+    </EuiButton>
+  );
+
+  const saveAndTestButton = (
+    <EuiButton
+      size="s"
+      iconType="eye"
+      onClick={handleSubmit(handleSaveAndTest)}
+      disabled={Object.keys(errors).length > 0 || isSubmitting}
+      isLoading={isSubmitting}
+      css={css`
+        width: 124px;
+      `}
+    >
+      {i18n.translate('xpack.onechat.tools.esqlToolFlyout.saveAndTestButtonLabel', {
+        defaultMessage: 'Save & test',
+      })}
+    </EuiButton>
+  );
+
+  const testButton = (
+    <EuiButton
+      size="s"
+      onClick={handleTestTool}
+      disabled={Object.keys(errors).length > 0}
+      iconType="eye"
+      css={css`
+        width: 124px;
+      `}
+    >
+      {i18n.translate('xpack.onechat.tools.esqlToolFlyout.testButtonLabel', {
+        defaultMessage: 'Test',
+      })}
+    </EuiButton>
+  );
+
+  return (
+    <FormProvider {...form}>
+      <KibanaPageTemplate>
+        <KibanaPageTemplate.Header
+          pageTitle={
+            [ToolFormMode.View, ToolFormMode.Edit].includes(mode)
+              ? tool?.id
+              : labels.tools.newToolTitle
+          }
+          css={css`
+            background-color: ${euiTheme.colors.backgroundBasePlain};
+            border-block-end: none;
+          `}
+        />
+        <KibanaPageTemplate.Section>
+          {isLoading ? (
+            <EuiFlexGroup justifyContent="center" alignItems="center">
+              <EuiLoadingSpinner size="xxl" />
+            </EuiFlexGroup>
+          ) : (
+            <>
+              {isViewMode ? (
+                <ToolForm mode={ToolFormMode.View} formId={toolFormId} />
+              ) : (
+                <ToolForm mode={mode} formId={toolFormId} saveTool={handleSave} />
+              )}
+              {showTestFlyout && currentToolId && (
+                <OnechatTestFlyout
+                  isOpen={showTestFlyout}
+                  isLoading={isLoading}
+                  toolId={currentToolId}
+                  onClose={() => {
+                    setShowTestFlyout(false);
+                    if (mode === ToolFormMode.Create) {
+                      navigateToOnechatUrl(appPaths.tools.list);
+                    }
+                  }}
+                />
+              )}
+            </>
+          )}
+        </KibanaPageTemplate.Section>
+        {!isViewMode && (
```

Comment:
not necessarly, we should be able to test system tools that are view only, they all follow standard schema so test flyout will work ;)